### PR TITLE
Update Firestore's public headers to use Swift class names.

### PR DESCRIPTION
**Pull request by ehsannas**
#no-changelog

### CODE REVIEW

**Comment by ehsannas** [comment]
#no-changelog

**Comment by ehsannas** [comment]
@ryanwilson  Not sure if it makes sense to also use `Dictionary` and `String` instead of `NSDictionary` and `NSString` in the documentation?

File: Firestore/Source/Public/FirebaseFirestore/FIRTimestamp.h [hunk 1]
```diff
@@ -59,15 +59,15 @@ NS_SWIFT_NAME(Timestamp)
...
 
 /**
  * Returns the result of comparing the receiver with another timestamp.
  * @param other the other timestamp to compare.
- * @return NSOrderedAscending if `other` is chronologically following self,
- *     NSOrderedDescending if `other` is chronologically preceding self,
- *     NSOrderedSame otherwise.
+ * @return `NSOrderedAscending` if `other` is chronologically following self,
+ *     `NSOrderedDescending` if `other` is chronologically preceding self,
+ *     `NSOrderedSame` otherwise.
```
Comment (ryanwilson) [inline Firestore/Source/Public/FirebaseFirestore/FIRTimestamp.h:70]: Looks like the Swift names are different: https://developer.apple.com/documentation/foundation/comparisonresult

**Comment by ryanwilson** [comment]
> @ryanwilson Not sure if it makes sense to also use `Dictionary` and `String` instead of `NSDictionary` and `NSString` in the documentation?

Yep, `Dictionary`, `String`, and `Array` all make sense as well in the docs. Thanks a ton for doing this!

File: Firestore/Source/Public/FirebaseFirestore/FIRTimestamp.h (hunk 1 above)
Comment (ehsannas) [reply Firestore/Source/Public/FirebaseFirestore/FIRTimestamp.h]: oops. Done.

**Review by ryanwilson — APPROVED** [review]
Awesome, thanks!

**Pull request merged**